src/snowflake/snowpark/session.py

```diff
@@ -725,8 +727,10 @@ def add_packages(
         :class:`~snowflake.snowpark.udf.UDFRegistration`. See details of
         `third-party Python packages in Snowflake <https://docs.snowflake.com/en/developer-guide/udf/python/udf-python-packages.html>`_.
 
-        Pure Python packages that are not available in Snowflake will be pip installed locally and made available as an
-        import (via zip file on a remote stage).
+        EXPERIMENTAL: Upon setting `upload_custom_packages` to True, Pure Python packages that are not available in
+        Snowflake will be pip installed locally and made available as an import (via zip file on a remote stage).
+        Note that if you wish to use a specific version of pip, you can set the environment variable `PIP_PATH` to your
+        pip executable. This feature is experimental and works well on UNIX systems only, do not use it in production!
```

Comment:
I wonder if we need to add a check for non-unix system (if possible at all).

Reply:
Added this as a warning -> https://github.com/snowflakedb/snowpark-python/pull/959/files#diff-9a82bd4a068fbfc98475b894a9901b9994eb8ca38b2cfda8f2b0427bc368749cR1033-R1036